api/src/main/java/uk/gov/dhsc/htbhf/claimant/entitlement/AdditionalPregnancyVoucherCalculator.java

```diff
@@ -0,0 +1,83 @@
+package uk.gov.dhsc.htbhf.claimant.entitlement;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Calculates the number of additional pregnancy vouchers when a claim has been updated to include an expected delivery date.
+ */
+@Component
+public class AdditionalPregnancyVoucherCalculator {
+
+    private final Integer entitlementCalculationDurationInDays;
+    private final Integer numberOfCalculationPeriods;
+    private final PregnancyEntitlementCalculator pregnancyEntitlementCalculator;
+    private final int vouchersPerPregnancy;
+
+    public AdditionalPregnancyVoucherCalculator(PaymentCycleConfig paymentCycleConfig,
+                                                PregnancyEntitlementCalculator pregnancyEntitlementCalculator,
+                                                @Value("${entitlement.number-of-vouchers-per-pregnancy}") int vouchersPerPregnancy) {
+        this.entitlementCalculationDurationInDays = paymentCycleConfig.getEntitlementCalculationDurationInDays();
+        this.numberOfCalculationPeriods = paymentCycleConfig.getNumberOfCalculationPeriods();
+        this.pregnancyEntitlementCalculator = pregnancyEntitlementCalculator;
+        this.vouchersPerPregnancy = vouchersPerPregnancy;
+    }
+
+    /**
+     * Get the number of additional pregnancy vouchers when we've been notified that a claimant is now pregnant.
+     * The pregnancy entitlement calculator is called for each full week that is left of the current cycle.
+     * e.g if the claim was updated mid way through week two of the cycle, the user will get pregnancy vouchers for week 3 and 4 (given a four week cycle).
+     * if the claim was updated during week four of the cycle, no additional vouchers will be returned.
+     *
+     * @param expectedDueDate       the expected due date of the claimant
+     * @param paymentCycleStartDate the start date for the current payment cycle
+     * @param claimUpdatedDate      the date that the claim was updated with the new expected due date
+     * @return the number of additional pregnancy vouchers the claimant is entitled to
+     */
+    public int getAdditionalPregnancyVouchers(LocalDate expectedDueDate,
+                                              LocalDate paymentCycleStartDate,
+                                              LocalDate claimUpdatedDate) {
+        if (claimUpdatedDate.isBefore(paymentCycleStartDate)) {
```

Comment:
Could we just handle this and the check below as
```
        if (!claimUpdatedDate.isAfter(paymentCycleStartDate)) {
            // if the update happened on the same day as the payment cycle start date,
            // then no need for additional vouchers as the payment cycle will now include pregnancy vouchers.
            return 0;
        }
```
?

Reply:
Replaced both checks with the one above.